### PR TITLE
TASK: Remove obsolete tables from new content repository

### PR DESCRIPTION
**Pull request by mficzel**
Add mysql migration to remove tables that are not used anymore in the cr.

**Review instructions**

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [x] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [x] Reviewer - The first section explains the change briefly for change-logs
- [x] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by bwaidelich** [comment]
Thanks for this, but I wonder: Instead of adding more migrations, couldn't we just remove the initial ones? This code was never released so IMO it does not make sense to create tables just to drop them immediately

**Comment by mficzel** [comment]
There are multiple migrations that created those tables that would have to be adjusted very carefully. It would be quite tedious to remove all those references.

**Comment by ahaeslich** [comment]
@bwaidelich I would be happy to rather have these migrations and be able to use them in our Neos 9 project instead of fixing the tables manually 😉.

File: Neos.Neos/Migrations/Mysql/Version20230307111325.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->abortIf(
+            !$this->connection->getDatabasePlatform() instanceof \Doctrine\DBAL\Platforms\MariaDb1027Platform,
+            "Migration can only be executed safely on '\Doctrine\DBAL\Platforms\MariaDb1027Platform'."
+        );
+
+        $this->addSql('DROP TABLE neos_neos_projection_asset_usage');
```
Comment (ahaeslich) [inline Neos.Neos/Migrations/Mysql/Version20230307111325.php]: Should we really drop this? I found the class `AssetUsageRepository` which uses this table. 🤔

Besides: The class would suggest that this migration should reside inside `Neos.ESCR.AssetUsage`.

**Review by ahaeslich — CHANGES_REQUESTED** [review]

File: Neos.Neos/Migrations/Mysql/Version20230307111325.php (hunk 1 above)
Comment (mficzel) [reply Neos.Neos/Migrations/Mysql/Version20230307111325.php]: You are right

Comment (mficzel) [reply Neos.Neos/Migrations/Mysql/Version20230307111325.php]: is adjusted

**Review by ahaeslich — APPROVED** [review]

**Review by bwaidelich — APPROVED** [review]
fair enough :)
+1 by reading

**Pull request merged**